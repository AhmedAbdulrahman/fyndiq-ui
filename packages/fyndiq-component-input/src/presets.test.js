import React from 'react'
import { shallow, mount } from 'enzyme'

import Presets from './presets'
import Input from './input'

const presets = ['test1', 'test2']

describe('fyndiq-component-input Presets', () => {
  it('should have the proper structure', () => {
    expect(shallow(<Presets presets={presets} />)).toMatchSnapshot()
  })

  it('should show a list of presets after clicking on the button', () => {
    const component = mount(<Presets presets={presets} />)
    component.find('Button').simulate('click')
    expect(component.find('ButtonWrapper')).toMatchSnapshot()
  })

  it('should call onChange if a preset is chosen', () => {
    const spy = jest.fn()
    const component = mount(<Presets presets={presets} onChange={spy} />)
    // Open the dropdown
    component.find('Button').simulate('click')
    // Click on the first button
    component
      .find('ButtonWrapper Button')
      .at(0)
      .simulate('click')

    expect(spy).toHaveBeenCalledWith('test1')
  })

  it('should call onChange if the input is modified', () => {
    const spy = jest.fn()
    const component = shallow(<Presets presets={presets} onChange={spy} />)

    component.find('Input').simulate('change', 'testValue')
    expect(spy).toHaveBeenCalledWith('testValue')
  })

  it('should have a custom input prop', () => {
    expect(
      mount(<Presets input={<Input className="inputClassName" />} />)
        .find('input')
        .hasClass('inputClassName'),
    ).toBe(true)
  })
})
