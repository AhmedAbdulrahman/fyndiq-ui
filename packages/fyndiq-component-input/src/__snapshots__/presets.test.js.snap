// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-input Presets should have the proper structure 1`] = `
<span
  className="presetWrapper"
>
  <Input
    className=""
    debounceWait={500}
    debouncedOnChange={null}
    inputRef={null}
    onChange={[Function]}
    type="text"
    value=""
  />
  <Dropdown
    button={
      <Button
        buttonRef={[Function]}
        className="presetDropdownButton"
        disabled={false}
        horizontal={false}
        htmlType="button"
        link={undefined}
        onClick={[Function]}
        pill={false}
        pressed={false}
        size="s"
        type="white"
      >
        <Arrow
          className="presetArrow"
          orientation="down"
        />
      </Button>
    }
    className=""
    disabled={false}
    hoverMode={false}
    margin={5}
    noArrow={false}
    noPropagateClickEvent={false}
    noWrapperStyle={false}
    onClose={[Function]}
    onOpen={[Function]}
    opened={false}
    position="br"
    wrapperClassName=""
  />
</span>
`;

exports[`fyndiq-component-input Presets should show a list of presets after clicking on the button 1`] = `
<ButtonWrapper
  className=""
  noOuterBorder={true}
  orientation="vertical"
>
  <div
    className="
      wrapper
      wrapper--orientation_vertical
      wrapperNoOuterBorder
      
    "
  >
    <Button
      buttonRef={[Function]}
      className="presetButton"
      disabled={false}
      horizontal={false}
      htmlType="button"
      key="test1"
      onClick={[Function]}
      pill={false}
      pressed={false}
      size="m"
      type="white"
    >
      <button
        className="
    button
    type-white
    size-m
    false
    false
    interactive
    false
    presetButton
  "
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        test1
      </button>
    </Button>
    <Button
      buttonRef={[Function]}
      className="presetButton"
      disabled={false}
      horizontal={false}
      htmlType="button"
      key="test2"
      onClick={[Function]}
      pill={false}
      pressed={false}
      size="m"
      type="white"
    >
      <button
        className="
    button
    type-white
    size-m
    false
    false
    interactive
    false
    presetButton
  "
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        test2
      </button>
    </Button>
  </div>
</ButtonWrapper>
`;
